### PR TITLE
Merge `53.0.0-dev` dev branch to main

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

# Rationale for this change
 
Now that we have made a  52.2.0  release candidate, https://github.com/apache/arrow-rs/issues/5688 we can introduce breaking API changes to main. 

Let's move our 53.0.0 development branch (see https://github.com/apache/arrow-rs/issues/6050 for backstory) to main

# What changes are included in this PR?

* bump tonic to 0.12 and prost to 0.13 for arrow-flight (#6041)
* Remove impl<T: AsRef<[u8]>> From<T> for Buffer that easily accident… https://github.com/apache/arrow-rs/pull/6043
* Make display of interval types more pretty (#6006) 
* Update snafu (#5930)
* Update Parquet thrift generated structures (#6045) 
* Revert "Revert "Write Bloom filters between row groups instead of the… 
* Update pyo3 requirement from 0.21.1 to 0.22.1 (fixed) (#6075) 
* remove repeated codes to make the codes more concise. (#6080)
* Add unencoded_byte_array_data_bytes to ParquetMetaData (#6068) 
* Update pyo3 requirement from 0.21.1 to 0.22.2 (#6085) 
* Deprecate read_page_locations() and simplify offset index in `ParquetMetaData` (#6095)
…

# Are there any user-facing changes?

Yes , see above

### CODE REVIEW

**Comment by etseidl** [comment]
merge oopsie 😮 

```diff
diff --git a/parquet/src/column/writer/mod.rs b/parquet/src/column/writer/mod.rs
index 643f34fbdd0..2c0c957d87d 100644
--- a/parquet/src/column/writer/mod.rs
+++ b/parquet/src/column/writer/mod.rs
@@ -298,7 +298,7 @@ impl<'a, E: ColumnValueEncoder> GenericColumnWriter<'a, E> {
                 num_page_nulls: 0,
             },
             column_metrics: ColumnMetrics::<E::T>::new(),
-            column_index_builder: ColumnIndexBuilder::new(),
+            column_index_builder,
             offset_index_builder: OffsetIndexBuilder::new(),
             encodings,
             data_page_boundary_ascending: true,
```

File: parquet/src/column/writer/mod.rs [hunk 1]
```diff
@@ -282,20 +297,8 @@ impl<'a, E: ColumnValueEncoder> GenericColumnWriter<'a, E> {
...
-                dictionary_page_offset: None,
-                data_page_offset: None,
-                min_column_value: None,
-                max_column_value: None,
-                num_column_nulls: 0,
-                column_distinct_count: None,
-            },
-            column_index_builder,
+            column_metrics: ColumnMetrics::<E::T>::new(),
+            column_index_builder: ColumnIndexBuilder::new(),
```
Comment (etseidl) [inline parquet/src/column/writer/mod.rs:301]:
```suggestion
            column_index_builder,
```

Comment (alamb) [reply parquet/src/column/writer/mod.rs:301]: Thank you --

**Review by etseidl — APPROVED** [review]
Thanks @alamb for taking this on! Looks good as far as my metadata changes are concerned.

**Comment by alamb** [comment]
Since this has all been reviewed previously I am going to merge it in now

**Pull request merged**